feathr_project/registry/registry/database.py

```diff
@@ -0,0 +1,84 @@
+from abc import ABC, abstractmethod
+import threading
+from distutils.log import debug, warn
+import os
+import pymssql
+
+
+providers = []
+
+class DbConnection(ABC):
+    @abstractmethod
+    def execute(self, sql: str, *args, **kwargs) -> list[dict]:
+        pass
+
+def quote(id):
+    if isinstance(id, str):
+        return f"'{id}'"
+    else:
+        return ",".join([f"'{i}'" for i in id])
+
+
+def parse_conn_str(s: str) -> dict:
+    """
+    TODO: Not a sound and safe implementation
+    """
+    parts = dict([p.strip().split("=", 1)
+                 for p in s.split(";") if len(p.strip()) > 0])
+    server = parts["Server"].split(":")[1].split(",")[0]
+    return {
+        "host": server,
+        "database": parts["Initial Catalog"],
+        "user": parts["User ID"],
+        "password": parts["Password"],
+        # "charset": "utf-8",   ## For unknown reason this causes connection failure
+    }
+
+
+class MssqlConnection(DbConnection):
+    @staticmethod
+    def connect(*args, **kwargs):
+        conn_str = os.environ["CONNECTION_STR"]
```

Comment:
update docs somewhere so end users know which environment to set. Also `CONNECTION_STR` sounds like a general one, maybe we should call it something specific to Feathr?

Reply:
It's just the generic format, not specific to Feathr.
This configuration is set in different ways for different deployment methods, e.g. AppService for Container and AKS have very different configuration mechanisms to deploy containers/pods.
@jainr is preparing the user manual and that should cover this part.